internal/ocm/tracing.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocm
+
+import (
+	"context"
+
+	arohcpv1alpha1 "github.com/openshift-online/ocm-sdk-go/arohcp/v1alpha1"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/Azure/ARO-HCP/internal/tracing"
+)
+
+type clusterServiceClientWithTracing struct {
+	csc        ClusterServiceClientSpec
```

Comment:
Consider embedding `ClusterServiceClientSpec` here so you don't have to do `csc.csc` everywhere.
```
type clusterServiceClientWithTracing struct {
        ClusterServiceClientSpec
        tracerName string
}
```